### PR TITLE
[FCE-495]: Create useStatus hook on mobile

### PR DESCRIPTION
**Pull request by MiloszFilimowski**
## Description

- Created the `usePeerStatus` hook, which represents the current connection status (WebSocket). We have the same hook on the web. 
- Refactored constants to enums on Android and iOS
- Refactored swift delegates/listeners into extensions to better visualize what is responsible for what. 

## Motivation and Context

- `usePeerStatus` was missing on mobile.

## How has this been tested?

Please describe in detail how you tested your changes. Include details of your
testing environment, devices (ex. Iphone XYZ ios X.X.X & Samsung XYZ android
X.X.X)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/react-native-client/src/hooks/usePeerStatus.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useCallback, useState } from 'react';
+import { PeerStatus } from '../types';
```
Comment (mironiasty) [inline packages/react-native-client/src/hooks/usePeerStatus.ts:2]: Why not define `PeerStatus` in this file?

File: packages/react-native-client/src/hooks/usePeerStatus.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    RNFishjamClientModule.peerStatus,
+  );
+
+  const onPeerStatusChanged = useCallback((status?: PeerStatus) => {
+    status && setPeerStatus(status);
+  }, []);
+
+  useFishjamEvent(ReceivableEvents.PeerStatusChanged, onPeerStatusChanged);
+
+  return peerStatus;
```
Comment (mironiasty) [inline packages/react-native-client/src/hooks/usePeerStatus.ts:23]: WDYT about returning `return { peerStatus }`?
So in unlikely event of adding new data to hook, we won't break backward compability.

Comment (MiloszFilimowski) [reply packages/react-native-client/src/hooks/usePeerStatus.ts:23]: Good idea 👍

File: packages/react-native-client/ios/Events.swift [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
+    case Reconnected
+    case Warning
+    case PeerStatusChanged
+
+    var name: String {
+        rawValue
+    }
+
+    static var allEvents: [String] {
+        EmitableEvents.allCases.map({ event in
```
Comment (maksg) [inline packages/react-native-client/ios/Events.swift:28]:
```suggestion
        EmitableEvents.allCases.map(\.name)
```

File: packages/react-native-client/src/RNFishjamClientModule.ts [hunk 4]
```diff
@@ -7,10 +7,17 @@ import type { CameraConfigInternal, Camera } from './hooks/useCamera';
...
+import { PeerStatus } from './hooks/usePeerStatus';
 
 type Metadata = { [key: string]: any };
 
 type RNFishjamClient = {
+  isMicrophoneOn: boolean;
+  isCameraOn: boolean;
+  isScreenShareOn: boolean;
+  cameras: ReadonlyArray<Camera>;
+  peerStatus: PeerStatus;
```
Comment (mironiasty) [inline packages/react-native-client/src/RNFishjamClientModule.ts:19]: nice 👍

**Review by mironiasty — APPROVED** [review]
LGTM

**Pull request merged**